initialization/initialization_test.go

```diff
@@ -868,3 +849,116 @@ func initData(datadir string) ([]byte, error) {
 
 	return io.ReadAll(reader)
 }
+
+func TestInitializeSubset(t *testing.T) {
+	r := require.New(t)
+	cfg := config.DefaultConfig()
+	cfg.LabelsPerUnit = 128
+
+	opts := config.DefaultInitOpts()
+	opts.DataDir = t.TempDir()
+	opts.NumUnits = 20
+	opts.MaxFileSize = cfg.LabelsPerUnit * 2 * uint64(config.BytesPerLabel()) // 2 units per file
+	opts.ProviderID = int(CPUProviderID())
+	opts.Scrypt.N = 2
+
+	init, err := NewInitializer(
+		WithNodeId(nodeId),
+		WithCommitmentAtxId(commitmentAtxId),
+		WithConfig(cfg),
+		WithInitOpts(opts),
+		WithLogger(zaptest.NewLogger(t, zaptest.Level(zap.DebugLevel))),
+	)
+	require.NoError(t, err)
+	err = init.Initialize(context.Background())
+	require.NoError(t, err)
+
+	optsSubset := opts
+	optsSubset.DataDir = t.TempDir()
+	optsSubset.FromFileIdx = 3
+	optsSubset.ToFileIdx = new(int)
+	*optsSubset.ToFileIdx = 4
+
+	initSubset, err := NewInitializer(
+		WithNodeId(nodeId),
+		WithCommitmentAtxId(commitmentAtxId),
+		WithConfig(cfg),
+		WithInitOpts(optsSubset),
+		WithLogger(zaptest.NewLogger(t, zaptest.Level(zap.DebugLevel))),
+	)
+	r.NoError(err)
+	err = initSubset.Initialize(context.Background())
+	r.NoError(err)
+
+	// Verify that the subset is a subset of the full set
+	fullData, err := initData(opts.DataDir)
+	r.NoError(err)
+	subsetData, err := initData(optsSubset.DataDir)
+	r.NoError(err)
+	r.True(bytes.Contains(fullData, subsetData))
```

Comment:
```suggestion
	r.Contains(fullData, subsetData)
```

Reply:
It doesn't work unless I wrap both in `string()` :shrug:

Reply:
I'm not blocking the merge because of this, just seems not necessary to me 🙂